docs/ingestion/compaction.md

```diff
@@ -82,7 +82,8 @@ If you configure query granularity in compaction to go from a finer granularity
 
 ### Dimension handling
 
-Apache Druid supports schema changes. Therefore, dimensions can be different across segments even if they are a part of the same data source. See [Different schemas among segments](../design/segments.md#different-schemas-among-segments). If the input segments have different dimensions, the resulting compacted segment include all dimensions of the input segments. 
+Apache Druid supports schema changes. Therefore, dimensions can be different across segments even if they are a part of the same data source. See [Different schemas among segments](../design/segments.md#segments-with-different-schemas). If the input segments have different dimensions, the resulting compacted segment include all dimensions of the input segments. 
+
```

Comment:
"include all dimensions of the input segments" --> "includes the union of all columns across all the input segments"

What happens if column `c` occurs in two input segments, but with differing types? Would be good to ask a Druid engineer what happens, then record that here.